### ISSUE TITLE
Auto-fill license field from license text

### ISSUE THREAD

**Issue by imphil** [librecores/librecores-web#70]
Match the license text to a set of well-known licenses.

Look into using SPDX and its associated match rules. SPDX is not complete for foss projects (OHDL, Solderpad are missing for example), can this be extended?

**Comment by imphil** [comment]
Python implementation: https://pypi.python.org/pypi/spdx-lookup

**Comment by imphil** [comment]
https://packagist.org/packages/composer/spdx-licenses has no matching implemented it seems

**Comment by imphil** [comment]
Maybe we can contribute that? Like `$licenses.guess($LICENSES_TEXT)`

**Comment by wallento** [comment]
Ups, @imphil was still logged on from the demo.

Maybe we can contribute that? Like `$licenses.guess($LICENSES_TEXT)`

**Comment by imphil** [comment]
Yes, porting the code from the python module should work. The SPEX rules to match a license text (https://spdx.org/spdx-license-list/matching-guidelines) are clear, but prose unfortunately.

**Comment by Agathver** [comment]
Hello, I would like to suggest looking at a ruby gem that does the job of guessing licenses from text (by using some math tricks). https://github.com/benbalter/licensee

This is [used by Github](https://help.github.com/articles/licensing-a-repository/#detecting-a-license) to detect licenses.

And additionally, by using the [Github license API](https://developer.github.com/v3/licenses/), licenses for Github repos can be fetched with no additional work.

**Comment by wallento** [comment]
Yes, that is the best candidate to guess. Actually, we were planning on restructuring the licenses a bit: There should be a list of known/established licenses to autocomplete the name of the license and the large textarea is only used then for unknown licenses. I am still not 100% sure how that looks best, but licensee would match into that by doing the initial guess.
